front/lib/api/files/upsert.ts

```diff
@@ -66,37 +67,45 @@ async function generateSnippet(
   const startTime = Date.now();
   const owner = auth.getNonNullableWorkspace();
 
-  switch (file.contentType) {
-    case "image/jpeg":
-    case "image/png":
-      return new Err(
-        new Error("Image files are not supported for file snippets.")
-      );
-    case "text/csv":
-    case "text/comma-separated-values":
-    case "text/tsv":
-    case "text/tab-separated-values":
-      const format =
-        file.contentType === "text/csv" ||
-        file.contentType === "text/comma-separated-values"
-          ? "csv"
-          : "tsv";
-
-      // Parse only the headers from the CSV file
-      const headers = content.split("\n")[0];
-
-      let snippet = `${format.toUpperCase()} file with headers: ${headers}`;
-      if (snippet.length > 256) {
-        snippet = snippet.slice(0, 242) + "... (truncated)";
-      }
+  if (isSupportedImageContentType(file.contentType)) {
+    return new Err(
+      new Error("Image files are not supported for file snippets.")
+    );
+  }
 
-      return new Ok(snippet);
-    case "text/markdown":
-    case "text/plain":
-    case "text/vnd.dust.attachment.slack.thread":
+  if (isSupportedDelimitedTextContentType(file.contentType)) {
+    const format =
+      file.contentType === "text/csv" ||
```

Comment:
this is a bit brittle? should we fetch the extension based on the content type since we capture that information

Reply:
We could also just use the content type here